### PR TITLE
many: rename AddAffected* to RegisterAffected*, add Change|State.Has, fix a comment

### PR DESCRIPTION
**Pull request by pedronis**
see each commit

the Has additions are both for completeness and for use in a future PR

### CODE REVIEW

File: overlord/state/change.go [hunk 1]
```diff
@@ -248,6 +248,12 @@ func (c *Change) Get(key string, value interface{}) error {
 	return c.data.get(key, value)
 }
 
+// Has returns whether the provided key has an associated value.
+func (c *Change) Has(key string) bool {
```
Comment (mardy) [inline overlord/state/change.go:252]: I have a slight preference for `hasKey`.

File: overlord/state/state.go [hunk 2]
```diff
@@ -271,6 +271,12 @@ func (s *State) Get(key string, value interface{}) error {
 	return s.data.get(key, value)
 }
 
+// Has returns whether the provided key has an associated value.
+func (s *State) Has(key string) bool {
```
Comment (mardy) [inline overlord/state/state.go:275]: same here

**Review by mardy — APPROVED** [review]
Thanks, LGTM!

The naming suggestion is a nitpick :-)

**Review by Meulengracht — APPROVED** [review]
LGTM!

**Pull request merged**